rfcs/0056-FE-APIs.md

```diff
@@ -0,0 +1,234 @@
+- Start Date: 2023-09-19
+- RFC PR: https://github.com/strapi/rfcs/pull/56
+
+# Summary
+
+The following document proposes new public facing APIs aimed at plugin developers to be able to augment the `EditView` of the content-manager, the need came from the design proposals of Draft & Publish and would most likely be implemented only in V5 of the product under the Draft & Publish scope.
+
+This document tries to perform the minimum required work to retain backwards compatible functionality yet deliver a solid developer experience and pave the way for more productive and intuitive public facing FE orientated APIs.
+
+# Example
+
+```jsx
+// admin/src/index.ts
+import type { DocumentAction } from "@strapi/admin/admin";
+import { PublishMultipleLocales } from "./documentActions";
+
+export default {
+  bootstrap(app) {
+    app.addDocumentActions([PublishMultipleLocales]);
+
+    app.addEditViewSidePanels([
+      {
+        title: "Review Workflows",
+        content: <ReviewWorkflowsSidePanel />,
+      },
+    ]);
+
+    app.addDocumentHeaderAction([
+      {
+        label: "Select Locale",
+        options: [{ label: "english", value: "en" }],
+      },
+    ]);
+  },
+};
+
+// admin/src/documentActions.ts
+export const PublishMultipleLocales = {
+  label: "Publish multiple locales",
+  onClick: () => window.alert("you're publishing!"),
+};
+```
+
+# Motivation
+
+Current injectionZones let you inject components into areas of the content-manager, while this is a good it's a somewhat limiting API, the `addDocumentHeaderAction` & `addEditViewSidePanels` APIs while the former may seem very constricted provides a balance between user's being able to add actions quicker but not affect the design of the product, the latter is incredibly flexible but a much clearer typed system compared to the current implementation of `InjectionZones`. The `addDocumentActions` API is a new API that allows you to add actions to be rendered in the DocumentActions menu (this is new to V5).
+
+Overall, we want to move towards more developer friendly APIs that are quicker to use, more intuitive yet do not cause concerns for the usability of our product.
+
+# Detailed proposal
```

Comment:
I am missing a information table in here so we can add stuff to this. https://gyazo.com/d40ae81220023684f1bc479fbd630b62